Dockerfile

```diff
@@ -0,0 +1,67 @@
+# syntax = docker/dockerfile-upstream:1.1.2-experimental
+
+# Meta args applied to stage base names.
+
+ARG TOOLS
+ARG GO_VERSION
+
+# The tools target provides base toolchain for the build.
+
+FROM $TOOLS AS tools
+ENV PATH /toolchain/bin:/toolchain/go/bin
+RUN ["/toolchain/bin/mkdir", "-p", "/bin", "/tmp"]
+RUN ["/toolchain/bin/ln", "-svf", "/toolchain/bin/bash", "/bin/sh"]
+RUN ["/toolchain/bin/ln", "-svf", "/toolchain/etc/ssl", "/etc/ssl"]
+RUN curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | \
+    bash -s -- -b /toolchain/bin v1.20.0
+RUN cd $(mktemp -d) \
+    && go mod init tmp \
+    && go get mvdan.cc/gofumpt/gofumports \
+    && mv /go/bin/gofumports /toolchain/go/bin/gofumports
+RUN curl -sfL https://github.com/uber/prototool/releases/download/v1.8.0/prototool-Linux-x86_64.tar.gz | \
+    tar -xz --strip-components=2 -C /toolchain/bin prototool/bin/prototool
+
+# The build target creates a container that will be used to build Talos source
+# code.
+
+FROM scratch AS build
+COPY --from=tools / /
+SHELL ["/toolchain/bin/bash", "-c"]
+ENV PATH /toolchain/bin:/toolchain/go/bin
+ENV GO111MODULE on
+ENV GOPROXY https://proxy.golang.org
+ENV CGO_ENABLED 0
+WORKDIR /src
+
+FROM build AS base
+COPY ./go.mod ./
+COPY ./go.sum ./
+RUN go mod download
+RUN go mod verify
+COPY ./pkg ./pkg
+COPY ./main.go ./main.go
+RUN go list -mod=readonly all >/dev/null
+RUN ! go mod tidy -v 2>&1 | grep .
+
+FROM base AS protoc-gen-proxy-build
+ARG SHA
+ARG TAG
+ARG VERSION_PKG="github.com/talos-systems/talos/pkg/version"
```

Comment:
Is this needed? This won't actually do anything in the current state.

Reply:
Copied from talos, but honestly there's no reason we should differ on the versioning. It's not currently included in the source but it's something that should get added. Probably a followup pr.